shells/lib/components/context/context-stores.js

```diff
@@ -8,56 +8,95 @@
 
 const ContextStoresImpl = class {
+  getStoreMetrics(store, isProfile) {
+    // TODO(sjmiles): store/handles are similar but different. We shoved the handle
+    // relating to this store onto the store object in ArcHandleListener. Probably
+    // this can be unified to use one or t'other.
+    const metrics = ContextStores.getHandleMetrics(store.handle, isProfile);
+    // TODO(sjmiles): roll this into 'metrics'? but thse are `store` not `handle` :(
+    metrics.backingStorageKey = store.storageKey;
+    if (store.backingStore) {
+      // TODO(sjmiles): property is not called 'storageKey' for pouchdb
+      metrics.backingStorageKey = store.backingStore.storageKey;
+    }
+    metrics.typeName = simpleNameOfType(store.type);
+    metrics.shareTypeName = `${metrics.typeName}Share`;
+    return metrics;
+  }
   getHandleMetrics(handle, isProfile) {
     const tag = (handle.tags && handle.tags.length) ? handle.tags.join('-') : null;
-    if (tag) {
-      const {base: userid, arcid, id} = crackStorageKey(handle.storageKey);
-      const type = handle.type.isCollection ? handle.type : handle.type.collectionOf();
-      const shareId = `${tag}|${id}|from|${userid}|${arcid}`;
-      const shortId = `${(isProfile ? `PROFILE` : `FRIEND`)}_${tag}`;
-      const storeName = shortId;
-      const storeId = isProfile ? shortId : shareId;
-      const boxStoreId = `BOXED_${tag}`;
-      const boxDataId = `${userid}|${arcid}`;
-      const metrics = {type, tags: [tag], storeId, storeName, boxStoreId, boxDataId};
-      return metrics;
-    }
+    const {base: userid, arcid, id} = crackStorageKey(handle.storageKey);
+    const type = handle.type.isCollection ? handle.type : handle.type.collectionOf();
+    const shareId = `${tag}|${id}|from|${userid}|${arcid}`;
+    const shortId = `${(isProfile ? `PROFILE` : `FRIEND`)}_${tag}`;
+    const storeName = shortId;
+    const storeId = isProfile ? shortId : shareId;
+    const boxStoreId = `BOXED_${tag}`;
+    const boxDataId = `${userid}|${arcid}`;
+    const metrics = {userid, type, tag, tags: [tag], storeId, storeName, boxStoreId, boxDataId};
+    return metrics;
   }
-  async getShareStore(context, type, name, id, tags) {
+  async getShareStore(context, schema, name, id, tags) {
     // cache and return promises in case of re-entrancy
     let promise = pendingStores[id];
     if (!promise) {
-      promise = new Promise(async (resolve) => {
+      promise = new Promise(async resolve => {
         const store = await context.findStoreById(id);
         if (store) {
           resolve(store);
         } else {
-          const store = await context.createStore(type, name, id, tags);
+          const store = await this.createReferenceStore(context, schema, name, id, tags);
           resolve(store);
         }
       });
       pendingStores[id] = promise;
     }
     return promise;
   }
-  getDecoratedId(entity, uid) {
-    return `${entity.id}:uid:${uid}`;
+  // getDecoratedId(entity, uid) {
```

Comment:
commented code here too.

Reply:
Fwiw, it's not uncommon for me to leave in commented code, at least, during this highly dynamic development phase. The code usually represents something I want to keep around for a cycle or two to make sure I don't actually need it. Having said that, in this case, I will remove it.